glicko2.py

```diff
@@ -146,28 +144,25 @@ def rate(self, rating, series):
             expected_score = self.expect_score(rating, other_rating, impact)
             variance_inv += impact ** 2 * expected_score * (1 - expected_score)
             difference += impact * (actual_score - expected_score)
-            d_square_inv += (
-                expected_score * (1 - expected_score) *
-                (Q ** 2) * (impact ** 2))
         difference /= variance_inv
-        variance = 1. / variance_inv
-        denom = rating.phi ** -2 + d_square_inv
-        phi = math.sqrt(1 / denom)
+        variance = 1.0 / variance_inv
         # Step 5. Determine the new value, Sigma', ot the sigma. This
         #         computation requires iteration.
         sigma = self.determine_sigma(rating, difference, variance)
         # Step 6. Update the rating deviation to the new pre-rating period
         #         value, Phi*.
-        phi_star = math.sqrt(phi ** 2 + sigma ** 2)
+        phi_star = math.sqrt(rating.phi ** 2 + sigma ** 2)
```

Comment:
We use the pre rating period RD value, as is demonstrated in the glicko 2 document by Glickman